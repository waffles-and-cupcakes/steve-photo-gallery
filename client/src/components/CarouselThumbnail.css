.thumbnailImage {
  height: 67px;
  width: 100px;
  cursor: pointer;
}