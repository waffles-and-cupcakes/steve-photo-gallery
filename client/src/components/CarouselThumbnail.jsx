import React from 'react';
import Slider from 'react-slick';
import './CarouselThumbnail.css';

const CarouselThumbnail = (props) => (
  <Slider
    accessability={true}
    swipe={false}
    arrows={false}
    focusOnSelect={true}
    centerMode={true}
    slidesToShow={9}
    asNavFor={props.main}
    ref={props.setThumbnail} >
    {props.photos.map((photo, index) => {
      return (
        <div>
          <img className="thumbnail-image" src={`${photo.thumbnailUrl}`} />
        </div>
      );
    })}
  </Slider>
);

export default CarouselThumbnail;