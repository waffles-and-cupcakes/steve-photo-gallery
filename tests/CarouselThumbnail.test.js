import React from 'react';
import { shallow, mount, render, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import CarouselThumbnail from '../client/src/components/CarouselThumbnail.jsx';

configure({adapter: new Adapter() });

const photos = [
  {
    title: 'A photo',
    isVerified: false,
    photoUrl: 'www.example.com/photo',
    thumbnailUrl: 'www.example.com/thumbnail',
    listingId: 1,
  },
  {
    title: 'Another photo',
    isVerified: true,
    photoUrl: 'www.example.com/photo',
    thumbnailUrl: 'www.example.com/thumbnail',
    listingId: 1,
  }
];

test('CarouselThumbnail should render correctly', () => {
  const output = shallow(<CarouselThumbnail photos={photos} />);
  expect(shallowToJson(output)).toMatchSnapshot();
});

test('CarouselThumbnail should render the number of slides based on windowWidth', () => {
  let output = shallow(<CarouselThumbnail photos={photos} windowWidth={300} />);
  expect(output.get(0).props.slidesToShow).toBe(1);

  output = shallow(<CarouselThumbnail photos={photos} windowWidth={700} />);
  expect(output.get(0).props.slidesToShow).toBe(3);

  output = shallow(<CarouselThumbnail photos={photos} windowWidth={900} />);
  expect(output.get(0).props.slidesToShow).toBe(5);

  output = shallow(<CarouselThumbnail photos={photos} windowWidth={1100} />);
  expect(output.get(0).props.slidesToShow).toBe(6);

  output = shallow(<CarouselThumbnail photos={photos} windowWidth={1400} />);
  expect(output.get(0).props.slidesToShow).toBe(7);

  output = shallow(<CarouselThumbnail photos={photos} windowWidth={1600} />);
  expect(output.get(0).props.slidesToShow).toBe(9);
});