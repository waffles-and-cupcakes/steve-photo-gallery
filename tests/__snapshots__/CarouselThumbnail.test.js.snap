// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselThumbnail should render correctly 1`] = `
<Slider
  accessability={true}
  arrows={false}
  centerMode={true}
  focusOnSelect={true}
  slidesToShow={9}
  swipe={false}
>
  <div
    key="www.example.com/thumbnail0"
  >
    <div>
      <img
        src="www.example.com/thumbnail"
      />
    </div>
  </div>
  <div
    key="www.example.com/thumbnail1"
  >
    <div>
      <img
        src="www.example.com/thumbnail"
      />
    </div>
  </div>
</Slider>
`;
